spec/unleash/configuration_spec.rb

```diff
@@ -126,14 +126,38 @@
       expect(config.http_headers).to include('UNLEASH-INSTANCEID')
     end
 
+    it "should allow lambdas and procs for custom_https_headers via new client" do
+      custom_headers_proc = proc do
+        { 'X-API-KEY' => '123' }
+      end
+      allow(custom_headers_proc).to receive(:call).and_call_original
+
+      config = Unleash::Configuration.new(
+        url: 'https://testurl/api',
+        app_name: 'test-app',
+        custom_http_headers: custom_headers_proc
+      )
+
+      expect{ config.validate! }.not_to raise_error
+      expect(config.custom_http_headers).to be_a(Proc)
+      expect(config.http_headers).to eq(
+        {
+          'X-API-KEY' => '123',
+          'UNLEASH-APPNAME' => 'test-app',
+          'UNLEASH-INSTANCEID' => config.instance_id
+        }
+      )
+      expect(custom_headers_proc).to have_received(:call)
```

Comment:
Nit picking again:

```suggestion
      expect(custom_headers_proc).to have_received(:call).twice
```
Or the correct number of times...

```suggestion
      expect(custom_headers_proc).to have_received(:call).exactly(2).times
```